azext_iot/operations/digitaltwin.py

```diff
@@ -0,0 +1,227 @@
+        raise CLIError('Command \'{}\' is not defined in the target interface.'.format(command_name))
+
+    target_command = [c for c in found_commands if c.get('name') == command_name]
+    if schema:
+        return target_command
+
+    if command_payload:
+        if exists(command_payload):
+            command_payload = str(read_file_content(command_payload))
+
+        target_json = None
+        try:
+            target_json = shell_safe_json_parse(command_payload)
+        except ValueError:
+            pass
+
+        if target_json or isinstance(target_json, bool):
+            command_payload = target_json
+
+    target = get_iot_hub_connection_string(cmd, hub_name, resource_group_name, login=login)
+    service_sdk, errors = _bind_sdk(target, SdkType.service_sdk)
+    try:
+        result = service_sdk.invoke_interface_command(device_id,
+                                                      interface,
+                                                      command_name,
+                                                      command_payload,
+                                                      connect_timeout_in_seconds=timeout,
+                                                      response_timeout_in_seconds=timeout)
+        return result
+    except errors.CloudError as e:
+        raise CLIError(unpack_msrest_error(e))
+
+
+def iot_digitaltwin_property_update(cmd, interface_payload, device_id,
+                                    hub_name=None, resource_group_name=None, login=None):
+    if exists(interface_payload):
+        interface_payload = str(read_file_content(interface_payload))
+
+    target_json = None
+    try:
+        target_json = shell_safe_json_parse(interface_payload)
+    except ValueError:
+        pass
+
+    if target_json:
+        interface_payload = target_json
+
+    target = get_iot_hub_connection_string(cmd, hub_name, resource_group_name, login=login)
+    service_sdk, errors = _bind_sdk(target, SdkType.service_sdk)
+    try:
+        result = service_sdk.update_interfaces(device_id, interfaces=interface_payload)
+        return result
+    except errors.CloudError as e:
+        raise CLIError(unpack_msrest_error(e))
+
+
+def iot_digitaltwin_monitor_events(cmd, device_id, interface, repo_endpoint=PNP_ENDPOINT, repo_id=None,
+                                   consumer_group='$Default', timeout=300, hub_name=None, resource_group_name=None,
+                                   yes=False, properties=None, repair=False, login=None, repo_login=None):
+    target_interfaces = []
+    pnp_context = {'enabled': True, 'interface': {}}
+    device_interfaces = _iot_digitaltwin_interface_list(cmd, device_id, hub_name, resource_group_name, login)
+    interface_list = _get_device_default_interface_dict(device_interfaces)
+    interface_count = len([k for k in interface_list if k['name'] == interface])
+    if interface and interface_count == 0:
```

Comment:
Regarding `interface_count` the list is already being iterated below extending `target_interfaces`. So move the statement on line# 168 up and then this check can be `if interface and not target_interfaces`.  Also is it a valid use case to have multiple interfaces with the same name? If so how is that supposed to work here? You monitor the same interface but different versions?

Reply:
fixed